cfpb/design-system
# Restructure DS website's information architecture

This PR restructures the site's navigation and moves every page into its new category. There's no longer a nav in the header, it's all in the sidebar now. It's also editable via the CMS to let users place pages exactly where they want instead of everything getting listed alphabetically.

We used to have a netlify cms [collection](https://github.com/cfpb/design-system/blob/ac5ee5e838543eb38e33c9b71401c39e0e36c70f/docs/admin/config.yml#L23) for every category but IMHO that's silly so I deleted them all and created a new collection called `pages` that houses every page.

## Changes

- Removes main nav and moves all nav elements to left-hand sidebar.
- Adds a side nav [data file](https://jekyllrb.com/docs/datafiles/) that is editable via the CMS and generates the left-hand sidebar.
- Adds a netlify cms collection called "Navigation" for the aforementioned new side nav data file.
- Removes the four netlify cms collections that were identical and replaces them with a single one called "Pages". Accordingly, I moved all website pages into a folder called `pages/`.
- Moves special pages like the homepage and the help page to a new collection called "Special pages".
- Updated the "Updating this website" page with instructions on how to update stuff.
- Removed the "Add new page button" and replaced it with a link in the footer.
- Updates tests and Jekyll/netlify metadata files as appropriate

## Testing

1. `yarn test:browser` should pass
1. Click around the PR preview link below and see how you feel about the new all-inclusive side nav.
1. Go to the "Updating this website" page by clicking the link in the footer and peruse the instructions for creating new pages and editing the side nav. **Note:** If you try and edit the side nav in the CMS on the PR preview website, it will be blank in the CMS because the `side-navigation.yml` file doesn't exist in this repo's master branch. You can run the site locally to preview it.

## Screenshots

| new side nav | editing side nav in CMS | new footer links |
|--------------|-------------------------|------------------|
| <img alt="screenshot" src="https://user-images.githubusercontent.com/1060248/85316075-1dffa080-b48a-11ea-8780-7042c96149c8.png"> | <img alt="screenshot" src="https://user-images.githubusercontent.com/1060248/85316080-1fc96400-b48a-11ea-9a44-61850381e26a.png"> | <img alt="screenshot" src="https://user-images.githubusercontent.com/1060248/85316078-1f30cd80-b48a-11ea-9d0b-660b08187aaa.png"> |



## Notes

-

## Todos

-

## Checklist


## Testing checklist

### Browsers


### Accessibility


### Other
